core-rust/state-manager/src/state_manager.rs

```diff
@@ -712,123 +711,132 @@ where
 
         // Whilst we could validate intent hash duplicates here, these are checked by validators on prepare already,
         // and the check will move into the engine at some point and we'll get it for free then...
-        let prepared_transactions: Vec<_> = commit_request
+        let prepare_results = commit_request
             .transactions
-            .into_iter()
-            .map(|raw| -> (RawLedgerTransaction, PreparedLedgerTransaction) {
-                let prepared = self.ledger_transaction_validator.prepare_from_raw(&raw)
-                    .unwrap_or_else(|error| {
-                        panic!("Committed transaction cannot be prepared - likely byzantine quorum: {error:?}");
-                    });
-                (raw, prepared)
+            .iter()
+            .map(|raw_transaction| {
+                self.ledger_transaction_validator
+                    .prepare_from_raw(raw_transaction)
             })
-            .collect();
+            .collect::<Result<Vec<_>, _>>();
+
+        let prepared_transactions = match prepare_results {
+            Ok(prepared_transactions) => prepared_transactions,
+            Err(error) => {
+                warn!("cannot parse transaction to be committed {error:?}",);
```

Comment:
Is it worth clarifying slightly more clearly that this is when validating a commit request?

It might be nice if this log included the node/source of the transaction, or maybe we already log that in the Java layer.

Reply:
> It might be nice if this log included the node/source of the transaction, or maybe we already log that in the Java layer.

Yes - the `Err`s are translated to exceptions, and caught, and logged with this detail: https://github.com/radixdlt/babylon-node/blob/33a9a5fd56b6e6dd0720e5e2bc54abc2f962f8e2/core/src/main/java/com/radixdlt/sync/LocalSyncService.java#L455-L460